### PR TITLE
Scratch smime tests

### PR DESCRIPTION
**Pull request by tomholub**
creating draft PR for easier review and discussion

### CODE REVIEW

File: conf/tsconfig.test.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "../test/source/patterns.ts",
     "../test/source/async-stack.ts",
     "../test/source/buf.ts",
+    "../test/source/smime-parse.ts",
```
Comment (tomholub) [inline conf/tsconfig.test.json:21]: You could skip this, instead put unit tests into `unit.ts` file

File: test/source/smime-parse.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+import * as ava from 'ava';
+import { SmimeKey } from '../../extension/js/common/core/crypto/smime/smime-key.js';
```
Comment (tomholub) [inline test/source/smime-parse.ts:4]: You can move this into `unit.ts` file and then import from a local core folder (symlink copy) in the test folder.

The "core" folder is the same, but the adjecent "platform" folder will provide platform specific stuff for the core to work on node.

File: test/source/smime-parse.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+x4B7FaD2dw0jTF/k5ulDmzUOc4k+j3LtZNbDOZMF/2g05hSKde/he1njlY3oKa9g
+VW8ftc2NwiSMthxyEIM+ALbNQVML2oN50gArBn5GeI22/aIBZxjtbEdmSTZIf82H
+sOwAnhJ+pD5iIPaF2oa0yN3PvI6IGxLpEv16tQO1N6e5bdP6ZDwqTQJyK+oNTNda
+yPLCqVTFJQWaCR5ZTekRQPTDZkjxjxbs
+-----END CERTIFICATE-----`;
+
+
+ava.default('smime-key parsing', async t => {
+  const key = await KeyUtil.parse(smimeCert);
+  expect(key.id).to.equal('63f7025e700f3945301fb2fba5674f84');
```
Comment (tomholub) [inline test/source/smime-parse.ts:40]: Should the id be capitalized? Or not for smime keys?

**Comment by wiktor-k** [comment]
Tom, this PR has been assimilated into #2739 including capitalization for S/MIME keys (for consistency) so I'll close it now.

**Pull request closed without merging**